realm/src/main/java/io/realm/RealmQuery.java

```diff
@@ -160,8 +153,15 @@ private boolean containsDot(String s) {
             if (columns.get(fieldName) == null) {
                 throw new IllegalArgumentException(String.format("Field '%s' does not exist.", fieldName));
             }
-            if (fieldType != table.getColumnType(columns.get(fieldName))) {
-                throw new IllegalArgumentException(String.format("Field '%s': type mismatch.", fieldName));
+
+            ColumnType tableColumnType = table.getColumnType(columns.get(fieldName));
+            if (fieldType != tableColumnType) {
+                throw new IllegalArgumentException(String.format(
+                        "Field '%s': type mismatch. Was %s, expected %s.",
+                        fieldName,
```

Comment:
This is a new "style" to split parameters with one new on each line. Seems very verbose.